src/include/miopen/generic_search.hpp

```diff
@@ -353,128 +353,134 @@ auto GenericSearch(const Solver s, const Context& context, const AnyInvokeParams
         compile_and_run = c_and_r;
     }
 
+    std::vector<KernelInfo> kernels;
     for(const auto& current_config : all_configs)
     {
-        float elapsed_time = 0.0f;
-        int ret            = 0;
-        MIOPEN_LOG_I2('#' << n_current << '/' << n_failed << '/' << n_runs_total << ' '
-                          << current_config);
+        ConvSolution current_solution = s.GetSolution(context, current_config, true);
+        for(auto&& kernel : current_solution.construction_params)
+        {
+            if(profile_h.HasProgram(kernel.kernel_file, kernel.comp_options))
+                continue;
+            kernels.push_back(kernel);
+        }
+    }
+    std::vector<Program> programs = PrecompileKernels(profile_h, kernels);
```

Comment:
If `BUILD_DEV=On`, then this is waste of time, because binary cache is switched OFF.

Reply:
is this a problem? we are aware that for this to work we need bin_cache turned on.

Reply:
Yes, because it affects all developers that work with auto-tuning. For example, recently I tried to test auto-tuning locally and found that it takes too long, so I just stopped trying.

Reply:
(I thought it was a compiler issue but actual reason is #307)

Reply:
We may want to add the runtime program handles to the KernelCache (at least for development builds). That would make the loop not redundant. Alternatively, we may disable the loop when the binary cache is disabled.

Reply:
i added this: #if MIOPEN_ENABLE_SQLITE_KERN_CACHE around the first loop, i can add a comment

Reply:
@alexandraBara Please do this
- Find IsCacheDisabled() in binary_cache.cpp
- Rename it to IsBinaryCacheDisabled()
- Make it global (remove static and add its declaration to binary_cache.hpp)
- Use it here instead of MIOPEN_ENABLE_SQLITE_KERN_CACHE